arviz/plots/pairplot.py

```diff
@@ -141,11 +143,20 @@ def plot_pair(
 
     # Get diverging draws and combine chains
     if divergences:
-        divergent_data = convert_to_dataset(data, group="sample_stats")
-        _, diverging_mask = xarray_to_ndarray(
-            divergent_data, var_names=("diverging",), combined=True
-        )
-        diverging_mask = np.squeeze(diverging_mask)
+        if hasattr(data, "sample_stats") and hasattr(data.sample_stats, "diverging"):
+            divergent_data = convert_to_dataset(data, group="sample_stats")
+            _, diverging_mask = xarray_to_ndarray(
+                divergent_data, var_names=("diverging",), combined=True
+            )
+            diverging_mask = np.squeeze(diverging_mask)
+        else:
+            divergences = False
+            warnings.warn(
+                "diverging field not found in sample stats. "
```

Comment:
Great message

Reply:
I like the message, but maybe someone may think that the diverging field _should_ be there. And most likely what is happening is that the sampling method does not provide that information.

Reply:
What about ```"Divergences data not found, plotting without divergences. Make sure the sample method provides divergences data and that it is present in the `diverging` field of `sample_stats` or set divergences=False"```